x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/xcontent/XContentUtils.java

```diff
@@ -65,4 +72,43 @@ public static String[] readStringArray(XContentParser parser, boolean allowNull)
         }
         return list.toArray(new String[list.size()]);
     }
+
+    /**
+     * Adds information about the permissions that a background service will run as to the X-Content representation
+     * of its configuration:
+     * - If the permissions are based on a user's roles at the time the config was created then the list of these
+     *   roles is added.
+     * - If the permissions come from an API key then the ID and name of the API key are added.
+     * - If the permissions come from a service account then the name of the service account is added.
+     * @param builder The {@link XContentBuilder} that the extra fields will be added to.
+     * @param headers Security headers that were stored to determine which permissions a background service
+     *                will run as. If <code>null</code> or no authentication key entry is present then no
+     *                fields are added.
+     */
+    public static void addStoredHeaderInfo(final XContentBuilder builder, final Map<String, String> headers) throws IOException {
+        if (headers == null) {
+            return;
+        }
+        String authKey = headers.get(AuthenticationField.AUTHENTICATION_KEY);
+        if (authKey == null) {
+            return;
+        }
+        builder.startObject(RoleDescriptor.Fields.RUN_AS.getPreferredName());
```

Comment:
I think it's better to find a different name for this field. `run_as` in RoleDescriptor as a specific meaning for user impersonation, which includes an authenticating user and an effective user. It is also a previlege that user can configure when creating a role. For the use case here, maybe something more generic like `authorization` could be better?